### ISSUE TITLE
Decrypting .maFile to access secret for KeePass

### ISSUE THREAD

**Issue by stoponerr** [Jessecar96/SteamDesktopAuthenticator#738]
Not an issue really, sorry, but I am not sure what parameters to use to try and decrypt the text of the file so I can set up TOTP for my password database without leaving the file in plaintext.

**Comment by dyc3** [comment]
If your maFiles are encrypted, you'll need to decrypt them with SDA. Then you can grab the `uri` field from the json. Should start with `otpauth://`. You should be able to import that into keepass.

**Comment by kazeXVI** [comment]
After setting up the TOTP with KeePass how can I stop using the SDA while still being able to use KeePass for codes? Obviously I can just only use KeePass but I would like KeePass to be the only place I get codes. Would deleting SDA and overwriting the data with something like BleachBIt be sufficent or would that cause problems if I somehow lost my KeePass database?

**Comment by Hrxn** [comment]
> If your maFiles are encrypted, you'll need to decrypt them with SDA. Then you can grab the `uri` field from the json. Should start with `otpauth://`. You should be able to import that into keepass.

Okay, I managed to decrypt my maFile within SDA, what do I have to do now? Do I need some kind of special option or extension for Keepass to import the `otpauth://` thing? 

@kazeXVI Well if you delete SDA completely one cannot generate OTPs this way, so I think this should be enough?
When setting up any Steam Authenticator, you'll get a revocation code, with this code (which you should've saved) you can always deactivate the Steam Authentication method..
But, obviously, you should never lose your Keepass database in the first place? I mean, that would be horrible. Without my Keepass database I'd be basically dead..

But one remaining question: How do trade confirmations work when using Keepass for code generation?!

**Comment by dyc3** [comment]
> But one remaining question: How do trade confirmations work when using Keepass for code generation?!

TL;DR: You don't.

Generating totp codes is standardized, and you can bring your secret around to different apps to generate codes. You don't need to disable or delete anything in SDA to generate codes from another app, as long as you are able to grab your shared secret.

Steam confirmations are a proprietary Valve thing, specific to Steam. If you can't make the specific HTTP calls to Valve's backend, you can't accept confirmations.
